feat: new twitter embed

cabljac opened the pull request:
This PR replaces the imported embedded tweet component with our own. There were formatting issues with the old component that were a pain, and this gives us more control.

This should fix https://github.com/invertase/docs.page/issues/190 , https://github.com/invertase/docs.page/issues/191 and https://github.com/invertase/docs.page/issues/192 .

Notes:

- It would be nice if aligning the tweets had some use on docs.page - perhaps to add a description and allow writiing alongside.

- A nicer loading state would be preferable

Ehesp commented on website/app/components/mdx/Tweet.tsx:
```diff
@@ -35,26 +38,43 @@ interface TweetProps {
+        })
+        .then(() => {
+          setIsLoading(false);
+          NProgress.done();
```
Isn't this going to show a loading spinner for the entire page if it's loading?

cabljac commented:
- I've tweaked the dark mode styles (twitter was giving us an ugly border that i've now removed.

- i've also introduced a dark mode context object to sync it with the dark mode state of the rest of the docs.
